server.js

```diff
@@ -1,34 +1,83 @@
-import express from 'express'
+import express, { request, response } from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+import listEndpoints from 'express-list-endpoints'
+
+import movies from './data/netflix-titles.json'
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
+// Middleware 
 app.use(cors())
-app.use(bodyParser.json())
+app.use(bodyParser.json()),
 
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+// Start route
+app.get('/', (request, response) => {
+  response.send(listEndpoints(app))
+})
+// return all movies 
+app.get('/movies', (request, response) => {
+  response.json(movies)
+})
+//return one single movie based on known ID
+app.get('/id/:id', (request, response) => {
+  const { id } = request.params 
+  const filteredMovies = movies
+    .find(movie => movie.show_id === +id
+  )
+filteredMovies ? response.json(filteredMovies) : response.status(404).send(`There is no movie with ID "${id}" in the system`)
+})
+//return movie based on year of release
+app.get('/year/:year', (request, response) => {
+  const year = request.params.year
+  const releaseYear = movies.filter(
+    item => item.release_year === +year
+  )
+  releaseYear.length !== 0
+    ?  
+      response.json({ length: releaseYear.length, data: releaseYear}) //json vs send?
+    :  
+      response.send({ length: releaseYear.length, data: releaseYear})
 })
+//return type of movies, filtered out to not show duplicates
+app.get('/type/type', (request, response) => {
```

Comment:
and remember that you need to use plural forms in your endpoints `/movies/types/:type`